src/types.ts

```diff
@@ -1475,6 +1475,179 @@ export type QuerySort<
   | SearchMessageSort<MessageType>
   | UserSort<UserType>;
 
+/**
+ * Client Rehydration Types
+ */
+
+export type ClientStateData<UserType = UnknownType> = {
+  userChannelReferences: { [key: string]: { [key: string]: boolean } };
+  users: {
+    [key: string]: UserResponse<UserType>;
+  };
+};
+
+export type ClientStateAndData<
```

Comment:
this type name is slightly confusing when looked at closely with the above, maybe `ClientStateAndUserData` would make more sense?